manifests/server/config.pp

```diff
@@ -57,6 +57,20 @@
   if ($logpath and $syslog) { fail('You cannot use syslog with logpath')}
 
   if ($ensure == 'present' or $ensure == true) {
+    # Install external mongo provider dependencies for ruby < 1.9.0
+    if versioncmp($::rubyversion, '1.9.0') < 0 {
+      # Active support provides ordered hashes and Ruby 1.8.7 must use ver 3.1.0 
+      package { 'activesupport':
+        ensure    => '3.1.0',
+        provider  => 'gem',
+      }
+      package { 'json':
```

Comment:
These packages have to only be optionally managed by the mongodb module, as they are not mongo-specific. Probably want to separate them into their own classes that may be optionally included (by the module with a true/false parameter, or externally).